plugins/outputs/dynatrace/dynatrace.go

```diff
@@ -267,17 +261,20 @@ func init() {
 	})
 }
 
-func getTypeOption(metricType telegraf.ValueType, field *telegraf.Field) dtMetric.MetricOption {
-	if metricType == telegraf.Counter {
-		switch v := field.Value.(type) {
-		case float64:
-			return dtMetric.WithFloatCounterValueTotal(v)
-		case uint64:
-			return dtMetric.WithIntCounterValueTotal(int64(v))
-		case int64:
-			return dtMetric.WithIntCounterValueTotal(v)
-		default:
-			return nil
+func (d *Dynatrace) getTypeOption(metric telegraf.Metric, field *telegraf.Field) dtMetric.MetricOption {
+	metricName := metric.Name() + "." + field.Key
+	for _, i := range d.AddCounterMetrics {
+		if metricName == i {
+			switch v := field.Value.(type) {
+			case float64:
+				return dtMetric.WithFloatCounterValueDelta(v)
+			case uint64:
+				return dtMetric.WithIntCounterValueDelta(int64(v))
+			case int64:
+				return dtMetric.WithIntCounterValueDelta(v)
+			default:
+				return nil
+			}
 		}
```

Comment:
You can save an indention level by
```suggestion
		if metricName != i {
			continue
		}
		switch v := field.Value.(type) {
		case float64:
			return dtMetric.WithFloatCounterValueDelta(v)
		case uint64:
			return dtMetric.WithIntCounterValueDelta(int64(v))
		case int64:
			return dtMetric.WithIntCounterValueDelta(v)
		default:
			return nil
		}
```